elastic/support-diagnostics
# Upgrading to Gradle build

**Overview**

- changing the project build tool from maven to gradle
- connects #490

**Notes**

- used `gradle init` command to update the project to gradle
- replaced all the maven plugins from pom.xml, with equivalent gradle task(s)
- deleted the maven assembly instructions (assembly.xml) and pom.xml files.
- no changes required on back-end 

**Testing Instructions**

- Ran the `gradle clean build` to verify that the build is working as expected
- verified that the `support-diagnostics-*.jar` file is getting generated

`Observations`
- Had to rely on SHA-256 checksum for verification, in place of SHA1, as the gradle plugin , [does not support SHA1 checksum generation as of today](https://github.com/gradle/gradle-checksum#usage).
- The jar file and other dependencies, and the test reports, are being assembled in the `build` directory now, instead of the `target` folder.
